api/proto/teleport/accesslist/v1/accesslist_service.proto

```diff
@@ -295,6 +299,69 @@ message UpsertAccessListWithMembersResponse {
   repeated Member members = 2;
 }
 
+// CreateAccessListWithPresetRequest is the request for creating
+// an access list with a preset.
+message CreateAccessListWithPresetRequest {
+  // access_list is the access list to create.
+  AccessList access_list = 1;
+  // members is the list of access list members to create.
+  repeated Member members = 2;
+  // preset_spec describes what type of preset was requested.
+  PresetSpec preset_spec = 3;
+}
+
+// CreateAccessListWithMembersAndPresetResponse is the response for creating
+// an access list with members and roles.
+message CreateAccessListWithPresetResponse {
+  // access_list is the access list that was created.
+  AccessList access_list = 1;
+  // members is the list of access list members that were created.
+  repeated Member members = 2;
+}
+
+// PresetType describes what type of preset was requested.
+enum PresetType {
+  // PRESET_TYPE_UNSPECIFIED is the zero value.
+  PRESET_TYPE_UNSPECIFIED = 0;
+  // PRESET_TYPE_LONG_TERM describes a preset where members are granted
+  // roles that grants long term access.
+  PRESET_TYPE_LONG_TERM = 1;
+  // GRANT_TYPE_SHORT_TERM describes a preset where members are granted
+  // roles that require them to request for access and upon approval gain short
+  // term access. And owners are granted roles that allows them to review those
+  // access requests.
+  PRESET_TYPE_SHORT_TERM = 2;
+}
+
+// PresetSpec defines the fields necessary for Teleport to perform actions
+// based on preset.
+message PresetSpec {
```

Comment:
I think the API is fine, but the separate "preset spec" makes it a little more confusing IMO. It's not really a preset, the API just needs to take an access list to create (w/ members), a set of roles to create and maybe a type to set as a label (preset type in your case IIUC). So I would consider simplifying and instead of separate preset spec just move type and roles into the request.

I think this is the same as Marek suggested [here](https://github.com/gravitational/teleport/pull/60078/files#r2428718145) basically.